wfe2/wfe.go

```diff
@@ -629,12 +629,36 @@ func (wfe *WebFrontEndImpl) NewAccount(
 		return
 	}
 
-	acct, err := wfe.RA.NewRegistration(ctx, core.Registration{
-		Contact:   accountCreateRequest.Contact,
-		Agreement: wfe.SubscriberAgreementURL,
-		Key:       key,
-		InitialIP: ip,
-	})
+	// Prepare account information to create corepb.Registration
+	keyBytes, err := key.MarshalJSON()
+	if err != nil {
+		wfe.sendError(response, logEvent,
+			web.ProblemDetailsForError(err, "Error creating new account"), err)
+		return
+	}
+	ipBytes, err := ip.MarshalText()
+	if err != nil {
+		wfe.sendError(response, logEvent,
+			web.ProblemDetailsForError(err, "Error creating new account"), err)
+		return
+	}
+	var contacts []string
+	contactsPresent := accountCreateRequest.Contact != nil
+	if accountCreateRequest.Contact != nil {
+		contacts = *accountCreateRequest.Contact
+	}
```

Comment:
The logic here is good. Since you're checking `.Contact != nil` twice, I would rearrange the code slightly to check it once- that makes it slightly easier for a reader to understand.


```suggestion
	var contacts []string
	var contactsPresent bool
	if accountCreateRequest.Contact != nil {
	        contactsPresent = true
 		contacts = *accountCreateRequest.Contact
	}
```